lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1315,7 +1295,7 @@ def deploy_vm(self):
         self.configure_cpu_and_memory(vm_obj=vm_obj, vm_creation=True)
         self.configure_disks(vm_obj=vm_obj)
         self.configure_network(vm_obj=vm_obj)
-        if self.should_deploy_from_template() and len(self.params['customize']) > 0:
+        if vm_obj is None or self.should_deploy_from_template():
```

Comment:
I prefer the second one. If you want to use customization (others than network customization) you need to set customization params.